sdk/identity/azure-identity/samples/client_certificate_credential.md

```diff
@@ -0,0 +1,69 @@
+
+### Explicit rotation
+
+If the application get's notified of certificate rotations and it can directly respond, it might choose to wrap the `CertificateCredential` in a custom credential which provides a means for rotating the certificate. 
+
+```py
+class RotatableCertificateCredential(object):
+    def __init__(self, tenant_id, client_id, certificate_path=None, **kwargs):
+        self._tenant_id = tenant_id
+        self._client_id = client_id
+        self._credential = CertificateCredential(tenant_id, client_id, certificate_path,**kwargs)
+    
+    def get_token(self, *scopes, **kwargs):
+        return self._credential.get_token(*scopes, **kwargs)
+
+    def rotate_certificate(certificate_path, **kwargs):
+        self._credential = CertificateCredential(self._tenant_id, self._client_id, certificate_path, **kwargs)
+```
+
+The above example shows a custom credential type `RotatableCertificateCredential` which provides a `rotate_certificate`. The implementation internally relies on a `CertificateCredential` instance `_credential`, and `rotate_certificate` simply replaces this instance with a new instance using the updated certificate.
+
+### Implicit rotation
+Some applications might want to respond to certificate rotations which are external to the application, for instance a separate process rotates the certificate by updating it on disk. Here the application create a custom credential which checks for certificate updates when tokens are requested. 
+
+```py
+class RotatingCertificateCredential(object):
+    def __init__(self, tenant_id, client_id, certificate_path=None, **kwargs):
+        self._tenant_id = tenant_id
+        self._client_id = client_id
+        self._certificate_path = certificate_path
+        self.refresh_certificate()
+
+    def get_token(self, *scopes, **kwargs):
+        self.refresh_certificate()
+        return self._credential.get_token(*scopes, **kwargs)
+
+    def refresh_certificate():
+        certificate_last_modified = os.path.getmtime(self._certificate_path)
+        if self._certificate_path < certificate_last_modified:
+            self._certificate_last_modified = certificate_last_modified
+            self._credential = CertificateCredential(tenant_id, client_id, certificate_path,**kwargs)
+```
+
+In this example the custom credential type `RotatingCertificateCredential` again uses a `CertificateCredential` instance `_credential` to retrieve tokens. However, in this case it will attempt to refresh the certificate prior to obtaining the token. The method `refresh_certificate` will query to see if the certificate has changed, and if so it will replace the instance `_credential` with a new instance using the new certificate.
```

Comment:
nit:
```suggestion
In this example the custom credential type `RotatingCertificateCredential` again uses a `CertificateCredential` instance, `_credential`, to retrieve tokens. However, in this case it will attempt to refresh the certificate prior to obtaining the token. The method `refresh_certificate` will query to see if the certificate has changed, and if so it will replace `_credential` with a new instance using the new certificate.
```